packages/lib/authOptions.ts

```diff
@@ -205,119 +211,104 @@ export const authOptions: NextAuthOptions = {
-                                      },
-                                      {
-                                        name: "50% Scroll",
-                                        description: "A user scrolled 50% of the current page",
-                                        type: "automatic",
-                                      },
-                                    ],
-                                  },
-                                  attributeClasses: {
-                                    create: [
-                                      {
-                                        name: "userId",
-                                        description: "The internal ID of the person",
-                                        type: "automatic",
-                                      },
-                                      {
-                                        name: "email",
-                                        description: "The email of the person",
-                                        type: "automatic",
-                                      },
-                                    ],
-                                  },
-                                },
-                              ],
-                            },
-                          },
-                        ],
-                      },
-                    },
-                  },
-                },
-              ],
-            },
-          },
-          include: {
-            memberships: true,
-          },
-        });
+            accounts: [{ ...account }],
+          });
+          const team = await createTeam({ name: userProfile.name + "'s Team" });
+          const product = await createProduct(team.id, { name: "My Product" });
+          const productionEnvironment = await createEnvironment(product.id, { type: "production" });
+          prodActionClasses.forEach(async (actionClass) => {
+            await createActionClass(productionEnvironment.id, {
+              ...actionClass,
+              environmentId: productionEnvironment.id,
+            });
+          });
+          prodAttributeClasses.forEach(async (attributeClass) => {
+            await createAttributeClass(productionEnvironment.id, attributeClass.name, attributeClass.type);
+          });
+          const developmentEnvironment = await createEnvironment(product.id, { type: "development" });
+          devActionClasses.forEach(async (actionClass) => {
+            await createActionClass(productionEnvironment.id, {
+              ...actionClass,
+              environmentId: developmentEnvironment.id,
+            });
+          });
+          devAttributeClasses.forEach(async (attributeClass) => {
+            await createAttributeClass(productionEnvironment.id, attributeClass.name, attributeClass.type);
+          });
+          await createMembership(team.id, user.id, { role: "owner" });
+        };
+
+        await registerUserAndSetupTeam();
```

Comment:
why did you put `registerUserAndSetupTeam()` into a function and didn't call it directly as it's only used in this place :-) 
I think if the function gets simpler without the dev and prod environment setup we can just put the function content here and have good readability